samza-core/src/main/java/org/apache/samza/processor/SamzaContainerController.java

```diff
@@ -60,16 +60,12 @@
    * @param taskFactory         Factory that be used create instances of {@link org.apache.samza.task.StreamTask} or
    *                            {@link org.apache.samza.task.AsyncStreamTask}
    * @param containerShutdownMs How long the Samza container should wait for an orderly shutdown of task instances
-   * @param processorId         Id of the processor
    * @param metricsReporterMap  Map of metric reporter name and {@link MetricsReporter} instance
    */
   public SamzaContainerController(
       Object taskFactory,
       long containerShutdownMs,
-      String processorId,
       Map<String, MetricsReporter> metricsReporterMap) {
-    this.executorService = Executors.newSingleThreadExecutor(new ThreadFactoryBuilder()
-        .setNameFormat("p" + processorId + "-container-thread-%d").build());
```

Comment:
- What is the cardinality of the relationship between `SamzaContainerController` and `SamzaContainer`? 

Some potential opportunities for clean-up:
- It seems that invoking `startContainer()` multiple times can potentially clobber the saved state in this class. I suspect we can add a guard to prevent this.
- Very minor: Are n't we creating multiple instances of `executorService` but only stopping the last created instance in `shutdown()` if `startContainer` is invoked multiple times? 

These are probably not blockers, so feel free to defer these for a future clean-up.